### DISCUSSION TITLE
Time depending cumulative function

### DISCUSSION THREAD

**Discussion by loixou** [google/or-tools#3273] [CP-SAT questions] [answered]
Good morning all,
I am wondering if there's a way to define a cumulative function which depends on time.
For instance, I would like:
```
0  <= t < T1, AddCumulative(intervals, 1, 10)
T1 <  t < T2, AddCumulative(intervals, 1, 5)
T2 <= t < Tf, AddCumulative(intervals, 1, 10)
```
I checked on Ortools documentation, but could not find anything about a capacity(t).
Thanks for helping :)

L.

**Comment by Mizux** [comment]
* #1210
* https://github.com/google/or-tools/issues/574#issuecomment-430203076
EDIT: my bad this is for routing solver not CP-SAT solver...

**Comment by lperron** [comment, marked as answer]
Just add fixed intervals and demands to lower the capacity from the max fixed capacity of the cumulative.

**Reply by loixou** [reply]
thanks ! :)
